lib/expression/v2.ex

```diff
@@ -84,6 +97,7 @@ defmodule Expression.V2 do
   def parse_block(expression_block) do
     case Parser.expression(expression_block) do
       {:ok, ast, "", _, _, _} -> {:ok, ast}
+      {:ok, _ast, _remainder, _, _, _} -> {:error, "Unable to parse", expression_block}
```

Comment:
When a block is invalid, the parser will return `{:ok, }` up until where it got to. For us this means the whole thing is still garbage and we should treat it as such.